### PR TITLE
#68: validate commit msg

### PR DESCRIPTION
**Pull request by artdaw**
Validate-commit-msg added as a commit message validator. Husky used for git hooks.

### CODE REVIEW

**Comment by artdaw** [comment]
👍

**Comment by ricardomatias** [comment]
I get this while trying to commit
`.git/hooks/pre-commit: line 2: ./node_modules/pre-commit/hook: No such file or directory`

**Comment by artdaw** [comment]
@ricardomatias remove all git hooks, node_modules and `npm install` again, please

File: package.json [hunk 1]
```diff
@@ -39,7 +39,9 @@
     "docs": "rm -rf docs && ./node_modules/.bin/jsdoc -r -c ./jsdoc_conf.json -t ./node_modules/ink-docstrap/template -d docs/ -R README.md",
     "docs:deploy": "./node_modules/.bin/gh-pages -d docs",
     "gh-pages-travis": "gh-pages-travis",
-    "commit": "./node_modules/.bin/git-cz",
+    "cz": "./node_modules/.bin/git-cz",
+    "precommit": "npm run validate",
```
Comment (ricardomatias) [inline package.json]: __IF__ we should lint and test, while using git, it should be done on pre-push.

Right now, if I write a bad message, it first lints and tests and only afterwards, it tells me that I didn't write the message correctly.

Comment (artdaw) [inline package.json]: @ricardomatias do you want to move validate to pre-push?

**Comment by artdaw** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

File: package.json (hunk 1 above)
Comment (ricardomatias) [inline package.json]: I don't see it in the code, did you push?

Comment (artdaw) [inline package.json]: Check again, please

**Comment by ricardomatias** [comment]
:+1:

**Comment by artdaw** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**